src/main/java/org/sigmah/server/handler/UpdateProjectHandler.java

```diff
@@ -163,21 +164,32 @@ public VoidResult execute(final UpdateProject cmd, final UserExecutionContext co
 	protected void updateProject(final List<ValueEventWrapper> values, final Integer projectId, UserExecutionContext context, String comment) throws CommandException {
 		// This date must be the same for all the saved values !
 		final Date historyDate = new Date();
-		
+
+		final User user = context.getUser();
+
 		// Search the given project.
 		final Project project = em().find(Project.class, projectId);
-		
+		if (project != null) {
+			if (!Handlers.isProjectEditable(project, user)) {
+				throw new IllegalStateException();
```

Comment:
Could you please add a message everytime you throw this exception (and a different one for each throw clause)? It would ease analyzing the stack-trace even if the code have moved (4 `throw` clause to fix).

I will update the coding guidelines for that.